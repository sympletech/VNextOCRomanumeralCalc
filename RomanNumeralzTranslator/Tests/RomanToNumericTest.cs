using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace RomanNumeralzTranslator
{
    [UseReporter(typeof(DiffReporter))]

    [TestClass]
    public class RomanToNumericTest
    {
        
        [TestMethod]
        public void RomanToDecimalTest()
        {
            string[] romanDigits = new[]
            {
                "I", "II", "III", "IV", "V", "VI", "VII", "VIII","IX","X",
                "XI", "XII", "XIII", "XIV", "XV", "XVI", "XVII", "XVIII","XIX","XX",
                "XXX","XXXIX","", "XL", "XLIX", "L", "l", "XC", "C",
                "XLVIII"
                
            };
            var results = romanDigits.Select(x => new {key = x, value = RomanNumeralTranslator.Translate(x)});
            ApprovalTests.Approvals.VerifyAll(results, "");
        }

        [TestMethod]
        public void RomanToDecimalExceptionTest()
        {
            var ex = ExceptionUtilities.GetException(() => RomanNumeralTranslator.Translate("Z"));
            Approvals.Verify(ex);
        }

        [TestMethod]
        public void BadDigitsTest()
        {
            string[] badDigits = {"IIX","IIZ","MIM", "MOM", "IVI", "HIV","XXVX", "VV","LL","LLCOOLJ"};
            var ex =
                badDigits.Select(
                    x =>
                        new
                        {
                            Key = x,
                            Value = ExceptionUtilities.GetException(() => RomanNumeralTranslator.Translate(x))
                        });
            Approvals.VerifyAll(ex,"");
        }
    }
}