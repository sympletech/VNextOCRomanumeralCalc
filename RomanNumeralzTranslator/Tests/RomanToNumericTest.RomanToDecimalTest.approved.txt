[0] = { key = I, value = 1 }
[1] = { key = II, value = 2 }
[2] = { key = III, value = 3 }
[3] = { key = IV, value = 4 }
[4] = { key = V, value = 5 }
[5] = { key = VI, value = 6 }
[6] = { key = VII, value = 7 }
[7] = { key = VIII, value = 8 }
[8] = { key = IX, value = 9 }
[9] = { key = X, value = 10 }
[10] = { key = XI, value = 11 }
[11] = { key = XII, value = 12 }
[12] = { key = XIII, value = 13 }
[13] = { key = XIV, value = 14 }
[14] = { key = XV, value = 15 }
[15] = { key = XVI, value = 16 }
[16] = { key = XVII, value = 17 }
[17] = { key = XVIII, value = 18 }
[18] = { key = XIX, value = 19 }
[19] = { key = XX, value = 20 }
[20] = { key = XXX, value = 30 }
[21] = { key = XXXIX, value = 39 }
[22] = { key = , value = 0 }
[23] = { key = XL, value = 40 }
[24] = { key = XLIX, value = 49 }
[25] = { key = L, value = 50 }
[26] = { key = l, value = 50 }
[27] = { key = XC, value = 90 }
[28] = { key = C, value = 100 }
[29] = { key = XLVIII, value = 48 }
